include/f_core/net/application/c_udp_broadcast_tenant.h

```diff
@@ -0,0 +1,56 @@
+#ifndef C_UDP_BROADCAST_TENANT_H
+#define C_UDP_BROADCAST_TENANT_H
+
+#include <f_core/net/network/c_ipv4.h>
+#include <f_core/net/transport/c_udp_socket.h>
+#include <f_core/messaging/c_message_port.h>
+
+template <typename T>
+class CUdpBroadcastTenant {
+public:
+    /**
+     * Constructor
+     * @param ipAddr Source IP address to broadcast from
+     * @param srcPort Source port to broadcast from
+     * @param dstPort Destination port to broadcast to
+     * @param messagePort Message port to receive messages to broadcast
+     */
+    CUdpBroadcastTenant(const char *ipAddr, const int srcPort, const int dstPort, CMessagePort<T> &messagePort) : udp(CIPv4(ipAddr), srcPort, dstPort), messagesToBroadcast(&messagePort) {}
+
+    /**
+     * Constructor
+     * @param udp UDP socket to broadcast messages to
+     * @param messagePort Message port to receive messages to broadcast
+     */
+    CUdpBroadcastTenant(const CUdpSocket& udp, CMessagePort<T> &messagePort) : udp(udp), messagesToBroadcast(&messagePort) {}
+
+    /**
+     * Destructor
+     */
+    ~CUdpBroadcastTenant() = default;
+
+    /**
+     * Synchronously transmit a message received from a message port over UDP
+     */
+    void TransmitMessageSynchronous() {
+        T message{};
+        if (messagesToBroadcast->Receive(message, K_FOREVER) == 0) {
+            udp.TransmitSynchronous(message, sizeof(T));
+        }
+    }
+
+    /**
+     * Asynchronously transmit a message received from a message port over UDP
+     */
+    void TransmitMessageAsynchronous() {
+        T message{};
+        if (messagesToBroadcast->Receive(message, K_NO_WAIT) == 0) {
+            udp.TransmitAsynchronous(message, sizeof(T));
+        }
+    }
+private:
+    CUdpSocket udp;
+    CMessagePort<T> *messagesToBroadcast;
```

Comment:
just a thought, who is resopnsible for deallocaitng a message port. i assume both the sender and reciever get a pointer to it to send and receive respectively but how who deletes it.

In practice we prolly wont be allocating those all the time but still feels odd to leak it.

Reply:
They're expected to last forever

Reply:
In future prs for fsw integrated into modules you'll see where they're stored. There's just one class that holds all the stuff we want to exist forever

Reply:
gotcha